iam_role.go

```diff
@@ -220,6 +249,121 @@ func (c *Client) CreateIamTrustRole(roleName string, roleType string, trustArn s
 	return cr, nil
 }
 
+type IamRoleInput struct {
+	RoleName                    *string            `json:"roleName"`
+	RoleType                    *string            `json:"roleType,omitempty"`
+	IncDefPols                  *int               `json:"includeDefaultPolicy,omitempty"`
+	AlksAccess                  *bool              `json:"enableAlksAccess,omitempty"`
+	TrustArn                    *string            `json:"trustArn,omitempty"`
+	TemplateFields              *map[string]string `json:"templateFields,omitempty"`
+	MaxSessionDurationInSeconds *int               `json:"maxSessionDurationInSeconds,omitempty"`
+	Tags                        *[]Tag             `json:"tags,omitempty"`
+}
+
+type IamRoleOutput struct {
+	BaseResponse
+	RoleName                    *string            `json:"roleName"`
+	RoleType                    *string            `json:"roleType,omitempty"`
+	RoleArn                     *string            `json:"roleArn,omitempty"`
+	RoleIPArn                   *string            `json:"instanceProfileArn,omitempty"`
+	RoleAddedToIP               *bool              `json:"addedRoleToInstanceProfile,omitempty"`
+	Exists                      *bool              `json:"roleExists,omitempty"`
+	TemplateFields              *map[string]string `json:"templateFields,omitempty"`
+	MaxSessionDurationInSeconds *int               `json:"maxSessionDurationInSeconds,omitempty"`
+	Tags                        *[]Tag             `json:"tags,omitempty"`
+}
+
+type encodeDetails struct {
+	*AccountDetails
+	*IamRoleInput
+}
+
+type requestOp struct {
+	ReqObj   []byte
+	Method   string
+	Endpoint string
+	Response *IamRoleOutput
+}
+
+func (enc *encodeDetails) MarshalJSON() ([]byte, error) {
+	obj, e := json.Marshal(*enc)
+	if e != nil {
+		return nil, e
+	}
+	return obj, e
+}
+
+func (c *Client) makeIamRoleRequest(operation *requestOp) error {
+	req, e := c.NewRequest(operation.ReqObj, operation.Method, operation.Endpoint)
+	if e != nil {
+		return e
+	}
+	resp, e := c.http.Do(req)
+	if e != nil {
+		return e
+	}
+
+	if e = decodeBody(resp, &operation.Response); e != nil {
+		if reqID := GetRequestID(resp); reqID != "" {
+			return fmt.Errorf("error parsing updateRole response: [%s] %s", reqID, e)
```

Comment:
Can this makeIamRoleRequest func be used for any operation? If so maybe we can update the errors based on what endpoint was called.

Reply:
definitely. I overlooked that error response.